### PR TITLE
Issue #113: Add reusable workflow for bytebase sql review pipeline

### PR DESCRIPTION
**Pull request by SonOfLope**
Since the workflow for validating changes with our bytebase policies will be used in different repositories, we are adding this reusable workflow with documentation.

will be called by ailab-db once this PR is merged : https://github.com/ai-cfia/ailab-db/pull/101
will be called by metadata-nachet-format once this PR is merged : https://github.com/ai-cfia/nachet-metadata-format/pull/17

### CODE REVIEW

File: .github/workflows/workflow-bytebase-sql-review.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+This workflow is used to review SQL changes in Bytebase. It is triggered by a
+pull request to the `main` branch of the calling repository.
+
+## Usage
+
+- **Purpose:** Parse and check the SQL statement according to the SQL review
+  rules set in our Bytebase instance to detect SQL anti-patterns and enforce
+  schema consistency across the organization.
+- **Secrets:** Secrets need to be configured from the calling repository.
+  - `BYTEBASE_API`: The API key used to authenticate with Bytebase.
```
Comment (Francois-Werbrouck) [inline .github/workflows/workflow-bytebase-sql-review.md:14]: We might need to be a little bit more specific for the origin of the API token

**Review by Francois-Werbrouck — APPROVED** [review]
LGTM

File: .github/workflows/workflow-bytebase-sql-review.md (hunk 1 above)
Comment (SonOfLope) [reply .github/workflows/workflow-bytebase-sql-review.md:14]: Done

**Comment by SonOfLope** [comment]
A fix will be added in [PR](https://github.com/ai-cfia/github-workflows/pull/111) with config file for dead links : 
{
    "aliveStatusCodes": [999,200,403]
}
to resolve the false positive deadlink error.

File: .github/workflows/workflow-bytebase-sql-review.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      - name: SQL advise
+        run: |
+          API="${{ secrets.BYTEBASE_API }}"
+          TOKEN="${{ secrets.SQL_REVIEW_API_SECRET }}"
+          echo "Start request $API"
+
+          pull_number=$(jq --raw-output .pull_request.number "$GITHUB_EVENT_PATH")
+          repository=`echo $GITHUB_REPOSITORY`
+          request_body=$(jq -n \
+          --arg repositoryId "$repository" \
```
Comment (rngadam) [inline .github/workflows/workflow-bytebase-sql-review.yml:20]: indent?

File: .github/workflows/workflow-bytebase-sql-review.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+          --arg pullRequestId $pull_number \
+          --arg webURL "$GITHUB_SERVER_URL" \
+          '$ARGS.named')
+
+          response=$(curl -s -w "%{http_code}" -X POST $API \
+          -H "X-SQL-Review-Token: $TOKEN" \
```
Comment (rngadam) [inline .github/workflows/workflow-bytebase-sql-review.yml:26]: indent?

File: .github/workflows/workflow-bytebase-sql-review.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+          if [ $http_code != 200 ]; then
+          echo ":error::Failed to check SQL with response code $http_code and body $body"
+          exit 1
+          fi
+
+          status=$(echo $body | jq -r '.status')
+          content=$(echo $body | jq -r '.content')
+
+          while read message; do
+          echo $message
```
Comment (rngadam) [inline .github/workflows/workflow-bytebase-sql-review.yml:43]: indent?

**Review by rngadam — APPROVED** [review]
which we could split off scripts to their own files instead of embedding it into yaml but there's then no really clean solution to get to them and execute them.

https://stackoverflow.com/questions/72133102/how-to-execute-a-a-remote-script-in-a-reusable-github-workflow

**Pull request merged**